### PR TITLE
optimize: clarify if conditions

### PR DESCRIPTION
**Pull request by yeoncheol-jang**
### Ⅰ. Describe what this PR did
(please see each commit)
- Changed to clarify the If condition that does various things
- Change the English grammar of comments to be more natural

### Ⅲ. Why don't you add test cases (unit test/integration test)? 
It has already tese cases.

### CODE REVIEW

**Comment by 1150456356** [comment]
good

**Comment by 1150456356** [comment]
good

**Comment by yeoncheol-jang** [comment]
cc. @funky-eyes

File: rm-datasource/src/main/java/org/apache/seata/rm/datasource/undo/AbstractUndoLogManager.java [hunk 1]
```diff
@@ -264,9 +264,10 @@ public void undo(DataSourceProxy dataSourceProxy, String xid, long branchId) thr
             try {
                 connectionProxy = dataSourceProxy.getConnection();
                 conn = connectionProxy.getTargetConnection();
+                originalAutoCommit = conn.getAutoCommit();
 
                 // The entire undo process should run in a local transaction.
-                if (originalAutoCommit = conn.getAutoCommit()) {
+                if (conn.getAutoCommit()) {
```
Comment (slievrly) [inline rm-datasource/src/main/java/org/apache/seata/rm/datasource/undo/AbstractUndoLogManager.java]: Sorry, I don't understand the purpose of doing this. Here, conn.getAutoCommit() is used twice.

Comment (yeoncheol-jang) [reply rm-datasource/src/main/java/org/apache/seata/rm/datasource/undo/AbstractUndoLogManager.java]: @slievrly Thank you for the review. 

I intended to divdie assign to variable and check whether is true or not in If condtion for clarify. `conn.getAutoCommit()` returns only state so i assumed will be okay for using twice.

What about these alternatively?
```
originalAutoCommit = conn.getAutoCommit();
if (originalAutoCommit)
```
or
```
final boolean autoCommitState = conn.getAutoCommit();
originalAutoCommit = autoCommitState
if (autoCommitState)
```

Comment (slievrly) [reply rm-datasource/src/main/java/org/apache/seata/rm/datasource/undo/AbstractUndoLogManager.java]: I prefer the first expression for its simplicity and fidelity to the original meaning.

**Comment by slievrly** [comment]
Additionally, remember to update the change log. This PR falls under the category of optimization.

File: rm-datasource/src/main/java/org/apache/seata/rm/datasource/undo/AbstractUndoLogManager.java (hunk 1 above)
Comment (yeoncheol-jang) [reply rm-datasource/src/main/java/org/apache/seata/rm/datasource/undo/AbstractUndoLogManager.java]: Thanks for your review! 
I've addressed your comments and pushed a new commit with the changes 😄

**Comment by yeoncheol-jang** [comment]
> Additionally, remember to update the change log. This PR falls under the category of optimization.

I changed commit message from 'refactor' to 'optimize'. Thanks for the review :)

**Comment by xingfudeshi** [comment]
@YeonCheolGit Hi,Please register your PR in those two files :
https://github.com/apache/incubator-seata/blob/2.x/changes/zh-cn/2.x.md 
https://github.com/apache/incubator-seata/blob/2.x/changes/en-us/2.x.md

**Comment by yeoncheol-jang** [comment]
> @YeonCheolGit Hi,Please register your PR in those two files : https://github.com/apache/incubator-seata/blob/2.x/changes/zh-cn/2.x.md https://github.com/apache/incubator-seata/blob/2.x/changes/en-us/2.x.md

Thank you for letting me know. I added my PR in these two files.
- https://github.com/apache/incubator-seata/pull/6464
- https://github.com/apache/incubator-seata/pull/6463

**Comment by xingfudeshi** [comment]
You should update the changes file in this PR without creating a separate PR.

**Comment by yeoncheol-jang** [comment]
> You should update the changes file in this PR without creating a separate PR.

Oh i misunderstanded. Add commit in this PR.

**Review by xingfudeshi — APPROVED** [review]
LGTM

File: changes/zh-cn/2.x.md [hunk 2]
```diff
@@ -115,6 +115,11 @@
 - [[#6387](https://github.com/apache/incubator-seata/pull/6387)] 优化tcc使用兼容
 - [[#6402](https://github.com/apache/incubator-seata/pull/6402)] 优化rm-datasource向下兼容
 - [[#6419](https://github.com/apache/incubator-seata/pull/6419)] 优化integration-tx-api向下兼容
+- [[#6442](https://github.com/apache/incubator-seata/pull/6442)] 阐明 if
```
Comment (slievrly) [inline changes/zh-cn/2.x.md:118]: The change log look there are some problems.

**Pull request merged**